hooks/useTokenList.tsx

```diff
@@ -23,7 +23,7 @@ export const useTokenList = () => {
   const { isLoading, error, data } = useSWR(
     `userTokenList-${chainId}`,
     async () => {
-      const resp = await fetch(network.tokenListUrl);
+      const resp = await fetch("tokenlist.json");
```

Comment:
This doesn't need to be async anymore, and probably doesn't need useSWR either? Or perhaps SWR is worth it since it's still a big file and a lot to process, to avoid re-processing when the component re-renders?

Reply:
hmm, after first request it should be cached by browser.
we shouldn't have to fetch this again when the chainId changes, because all chains read from the same list. maybe we could store the initial load to state, and then reactively filter when the network changes.

Reply:
Should we use useSWRImmutable instead?

https://swr.vercel.app/blog/swr-v1#immutable-mode

Reply:
ah yes forgot about that, yea we should

Reply:
Cool, I split out the fetch and used the immutable hook.

Reply:
sweet, but why do we still do `await fetch("tokenlist.json")` instead of just importing the JSON? seems unnecessary to `fetch` it and have two async calls here

Reply:
Fetching prevents it from being in the bundle the difference  isn't huge though. Running build with a fetch vs importing the json file looks like below.

With import 
![Screenshot from 2023-01-24 15-28-06](https://user-images.githubusercontent.com/12705146/214401695-dd9c04d0-9358-48a8-b576-05fc165e04dd.png)

With fetch
![Screenshot from 2023-01-24 15-28-34](https://user-images.githubusercontent.com/12705146/214401717-6e8cc7b9-f7fb-4e76-b0c0-1a2c55cc462a.png)

Reply:
I am open to importing if everyone is good with  little bundle bloat.

Reply:
> Fetching prevents it from being in the bundle the difference isn't huge though

Ah interesting.

Personally I prefer the simpler code and 5kb extra of bundle size, since that's pretty negligible compared to a 2.2MB bundle, and that list need to be loaded to use the app no matter what anyway